project-base/storefront/components/Basic/ModalGallery/ModalGallery.tsx

```diff
@@ -0,0 +1,212 @@
+import { PlayIcon } from 'components/Basic/Icon/IconsSvg';
+import { Image } from 'components/Basic/Image/Image';
+import { VideoTokenFragmentApi, ImageFragmentApi } from 'graphql/generated';
+import { twMergeCustom } from 'helpers/twMerge';
+import { useKeypress } from 'hooks/useKeyPress';
+import useTranslation from 'next-translate/useTranslation';
+import { RefObject, createRef, useEffect, useState } from 'react';
+import { useSwipeable } from 'react-swipeable';
+import { twJoin } from 'tailwind-merge';
+
+type ModalGalleryProps = {
+    galleryName: string;
+    initialIndex: number;
+    items: (VideoTokenFragmentApi | ImageFragmentApi)[];
+    onCloseModal: () => void;
+};
+
+export const ModalGallery: FC<ModalGalleryProps> = ({ initialIndex, items, galleryName, onCloseModal }) => {
+    const { t } = useTranslation();
+
+    const [selectedIndex, setSelectedIndex] = useState(initialIndex);
+    const [itemsRefs, setItemsRefs] = useState<Array<RefObject<HTMLLIElement>>>();
+    const [isLoaded, setIsLoaded] = useState(false);
+
+    const selectedGalleryItem = items[selectedIndex];
+
+    const isImage = selectedGalleryItem.__typename === 'Image';
+    const isVideo = selectedGalleryItem.__typename === 'VideoToken';
+
+    const lastItemIndex = items.length - 1;
+
+    const isCarouselDisplayed = items.length > 1 && (!isImage || isLoaded);
+
+    const selectPreviousItem = () => {
+        setSelectedIndex(selectedIndex > 0 ? selectedIndex - 1 : lastItemIndex);
+    };
+
+    const selectNextItem = () => {
+        setSelectedIndex(selectedIndex < lastItemIndex ? selectedIndex + 1 : 0);
+    };
```

Comment:
Just a suggestion if you dont want to manipulate the current value.

Reply:
Hmm, I don't understand what you mean by current value? 🤔

Reply:
I meant setSelectedIndex(currentValue => ...). Since there can be multiple calls to this method in a render (hypothetically), it might operate on an older value. Especially since this is a relative operation (increase decrease of the current value), this might be a safer way to approach it.

Reply:
Yup, good idea!